pkg/pgclient/client.go

```diff
@@ -105,31 +110,34 @@ func NewClient(cfg *Config, schemaLocker LockFunc) (*Client, error) {
 }
 
 // NewClientWithPool creates a new PostgreSQL client with an existing connection pool.
-func NewClientWithPool(cfg *Config, numCopiers int, pool *pgxpool.Pool) (*Client, error) {
-	cache := &pgmodel.MetricNameCache{Metrics: clockcache.WithMax(cfg.MetricsCacheSize)}
-
+func NewClientWithPool(cfg *Config, numCopiers int, dbConn pgxconn.PgxConn) (*Client, error) {
+	metricsCache := &pgmodel.MetricNameCache{Metrics: clockcache.WithMax(cfg.MetricsCacheSize)}
+	labelsCache := clockcache.WithMax(cfg.LabelsCacheSize)
 	c := pgmodel.Cfg{
 		AsyncAcks:       cfg.AsyncAcks,
 		ReportInterval:  cfg.ReportInterval,
 		SeriesCacheSize: cfg.SeriesCacheSize,
 		NumCopiers:      numCopiers,
 	}
-	ingestor, err := pgmodel.NewPgxIngestorWithMetricCache(pool, cache, &c)
+	ingestor, err := pgmodel.NewPgxIngestorWithMetricCache(dbConn, metricsCache, &c)
 	if err != nil {
 		log.Error("msg", "err starting ingestor", "err", err)
 		return nil, err
 	}
-	reader := pgmodel.NewPgxReaderWithMetricCache(pool, cache, cfg.LabelsCacheSize)
+	querier := pgmodel.NewQuerierWithCaches(dbConn, metricsCache, labelsCache)
 
-	queryable := query.NewQueryable(reader.GetQuerier())
+	queryable := query.NewQueryable(querier)
 
+	healthChecker := pgmodel.NewHealthChecker(dbConn)
 	client := &Client{
-		Connection:  pool,
-		ingestor:    ingestor,
-		reader:      reader,
-		queryable:   queryable,
-		cfg:         cfg,
-		metricCache: cache,
+		Connection:    dbConn,
```

Comment:
it shouldn't be exposed or returned now since the connection here is given as an arugment, but it is required when creating the PgDelete and will require refactoring that I will not put in this PR